File: Go/pointers.go
// Example of a pointer in Go
package main

import "fmt"

func main() {
	// Declare a variable and assign a value, and print its address
	var number int = 10
	fmt.Printf("Number: %d, Address: %p\n", number, &number)

	// Declare a pointer to an integer, and assign the address of the variable
	var ptr *int = &number
	fmt.Printf("Pointer: %p, Value: %d\n", ptr, *ptr)
	
  // Change the value through the pointer
	*ptr = 20
	fmt.Printf("Number after change: %d\n", number)
	
  // Dereference the pointer to get the original value
	fmt.Printf("Dereferenced pointer: %d\n", *ptr)
}

File: Rust/pointers.rs
// Example of pointers in Rust
fn main() {
    // Declare a mutable integer variable.
    let mut x = 5;

    // Create a pointer to the integer variable.
    let ptr_x = &mut x;

    // Change the value through the pointer.
    *ptr_x = 10;

    // Print the value through the pointer.
    println!("The value of x is: {}", x);

    // Declare a reference to a reference to an integer variable.
    let ptr_ptr_x = &ptr_x;

    // Change the value through the reference.
    **ptr_ptr_x = 20;

    // Print the value through the reference.
    println!("The value of x is: {}", x);
}
